File: 01137. N-th Tribonacci Number/01137.c
int tribonacci(int n){
    static int arr[38] = {0, 1, 1, 2};
    
    if (n >= 4 && arr[n] == 0) {
        for (int i=4, count=2; i<38; ++i) {
            count += arr[i-1] - arr[i-4];
            arr[i] = count;
        }
    }
    
    return arr[n];
}

File: LeetCode/01137. N-th Tribonacci Number/01137.go
// https://leetcode.com/problems/n-th-tribonacci-number

var once bool
var arr [38]int

func tribonacci(n int) int {
    if once == false {
        once = true

        arr[0], arr[1], arr[2] = 0, 1, 1
        for i := 3; i < 38; i++ {
            arr[i] = arr[i-3] + arr[i-2] + arr[i-1]
        }
    }

    return arr[n]
}
